src/jit/lclvars.cpp

```diff
@@ -3292,6 +3292,13 @@ void Compiler::lvaSortByRefCount()
         {
             varDsc->lvTracked = 0;
         }
+#if defined(JIT32_GCENCODER) && defined(WIN64EXCEPTIONS)
+        else if (lvaIsOriginalThisArg(lclNum))
+        {
+            // For x86/Linux, we cannot have this pointer in tracked variables, so we set this pointer untracked
+            varDsc->lvTracked = 0;
```

Comment:
@BruceForstall I agree. As I mentioned in https://github.com/dotnet/coreclr/pull/11281#issuecomment-302983448 I'll update it.